### ISSUE TITLE
--maintain-node-health does not work on two node cluster

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#801]
Originally reported of Google Code with ID 745.

```
What software version are you running? Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".

gnt-cluster (ganeti v2.9.5) 2.9.5

Software version: 2.9.5
Internode protocol: 2090000
Configuration format: 2090000
OS api version: 20
Export interface: 0
VCS version: v2.9.5


<b>What distribution are you using?</b>

ubuntu 12.04.4 LTS


<b>What steps will reproduce the problem?</b>
1. on a two node cluster enable --maintain-node-health
2. set second node offline

<b>What is the expected output? What do you see instead?</b>

ganeti-watcher should stop instances and disable drbd devices.


<b>Please provide any additional information below.</b>

ganeti-watcher just shows a warning:

2014-02-25 10:42:23,062: ganeti-watcher pid=17690 nodemaint:146 WARNING Inconsistent replies, not doing anything

i also get this on a second two node cluster with same os and ganeti versions.

see discussion -> https://groups.google.com/forum/#!topic/ganeti/oSqvQJ8oysg
```
Originally added on 2014-03-04 07:16:54 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I just tried this on a newly-installed test cluster with 2.9.5 and I was unable to reproduce: no WARNING is shown in the logs.

Could you whether it is visible on a newly-installed cluster and report the exact sequence of steps to reproduce the bug?
```
Originally added on 2014-03-04 12:38:45 +0000 UTC.

Changed State: NeedFeedback

**Comment by Ganeti-Issues-Migrator** [comment]
```
i'm sorry but both are production clusters and i cannot re-create a cluster on those machines.

if there is anything else i can do to debug this issue let me know.

is you test cluster a two node cluster? did you try to run ganeti-watcher -d on the offline node? this is what i did to get this warning message.
```
Originally added on 2014-03-04 14:03:10 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Ah! I hadn't realized that the error message was supposed to appear on the node marked as offline!

Yes, the warning is indeed visible in watcher.log on the offline node, and running "xm list" on that node (on a xen cluster) shows that the instance is indeed still running.

Bug confirmed, thanks for the report.
```
Originally added on 2014-03-04 14:14:54 +0000 UTC.

Added Labels: Type-Defect Priority-Medium
Changed State: Accepted
Added to Milestone: Release2.9

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit fa3c0df5726a203e54eb66b117b16c5da3ca1231
Author: Klaus Aehlig <aehlig@google.com>
Date:   Fri Jul 31 14:46:34 2015 +0200

    In confd provide proper serial number
    
    The confd protocol heavily relies on the serial number
    to filter out outdated responses. However, the current
    implementation always returned 0 as serial number. Fix
    this and return a serial number that is bumped with every
    change that affects the answer.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

```
Originally added on 2015-07-31 13:32:58 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
fix pushed to 2.12
```
Originally added on 2015-07-31 13:33:41 +0000 UTC.

Changed State: Fixed
Added to Milestone: Release2.12
Removed from Milestone: Release2.9
